skip initramfs actors when livemode is enabled

PeterMocary opened the pull request:
The new storage initialization solution interfered with livemode. Since livemode is a different approach to the upgrade, the initramfs actors (enable_lvm_autoactivation and mount_unit_generator) related to the new storage initialization are now skipped when upgrading via this method.

Jira: RHEL-119516

pirat89 reviewed (changes requested):
it needs some changes. see my comments above.

matejmatuska commented:
I did a test on both rhel8 and rhel9, without this patch the upgrade fails at the start of initramfs, with it works allright.

PeterMocary committed:
skip pre-generation of systemd fstab mount units during LiveMode upgrade

The new storage initialization solution interfered with LiveMode. Since
LiveMode is a different upgrade approach, we now skip pre-generation of
systemd fstab mount units (mount_unit_generator actor) when
upgrading this way.

PeterMocary commented on repos/system_upgrade/common/actors/initramfs/enable_lvm_autoactivation/libraries/enable_lvm_autoactivation.py:
```diff
@@ -1,9 +1,16 @@
+        api.current_logger().debug(
+            'Skipping LVM autoactivation instructions production due to livemode being enabled.'
+        )
+        return
```
I see, so the lvm auto activation actor is no longer skipped in the changes.

PeterMocary commented:
Updated the changes according to the review from @pirat89.

PeterMocary commented:
Rebased against upstream.

pirat89 reviewed (approved):
lgtm!

pirat89 commented:
/packit copr-build
